Modules/Sources/Yosemite/Model/Storage/PersistedProduct+Conversions.swift

```diff
@@ -42,11 +42,47 @@ extension PersistedProduct {
             stockStatusKey: stockStatusKey
         )
     }
+
+    func toPOSProduct(db: GRDBDatabaseConnection) throws -> POSProduct {
+        let images = try db.read { db in
+            return try request(for: PersistedProduct.images).fetchAll(db)
+        }
+        let attributes = try db.read { db in
+            return try request(for: PersistedProduct.attributes).fetchAll(db)
+        }
```

Comment:
nit: could these be combined into one database transaction, potentially saving some overhead from each `db.read`:

```suggestion
        let (images, attributes) = try db.read { db in
            let images = try request(for: PersistedProduct.images).fetchAll(db)
            let attributes = try request(for: PersistedProduct.attributes).fetchAll(db)
            return (images, attributes)
        }
```

Reply:
Both done in e3d7b85